app/controllers/application_controller.rb

```diff
@@ -44,4 +44,15 @@ def store_location
       session[:return_to] = request.original_fullpath
     end
   end
+
+  def store_referer
```

Comment:
ok cool - maybe in the future we can reconstruct the requests so that they can share logic since these seem highly related